### PR TITLE
Added google login backend route and integrated it with frontend

### LINKED ISSUE #183: Adding google-login route on backend and integrating it with frontend
## Describe the bug/feature

A user logging in using google is not being stored in database having some basic details currently. so he is not able to edit his profile , we can't store projects added by him or messages sent by him without storing user with required fields.

@rockingrohit9639 I'd like to work on this issue,

### PR DESCRIPTION
**Pull request by ShwetKhatri2001**
### Description

- User trying google signin/signup will be signed up and stored in database , if not done before , else will be logged in and previously stored / updated details will be fetched and shown on front-end.

Fixes #183

### Type of Change:
- New Feature

### How Has This Been Tested?
User is checked to be saved in database and  being updated by checking edit profile / add new project. After every login previous data is being fetched correctly.

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: server/routes/projectRoutes.js [hunk 1]
```diff
@@ -38,30 +38,6 @@ projectsRouter.get("/getprojects", async (req, res) => {
...
-      return res
-        .status(200)
-        .json({ message: "Successully added your project." });
-    } else {
-      return res.status(400).json({ error: "Could not add your project." });
-    }
-  } catch (err) {
-    return res.status(400).json({ error: "Could not add your project." });
-  }
-});
```
Comment (satyampgt4) [inline server/routes/projectRoutes.js:64]: Any specific reason for removing this part...?

**Review by satyampgt4 — APPROVED** [review]
LGTM

File: server/routes/projectRoutes.js (hunk 1 above)
Comment (rockingrohit9639) [reply server/routes/projectRoutes.js:64]: it was unnecessary and was blocking another request.

**Comment by kanak22** [comment]
Kindly resolve the merge conflicts

**Pull request merged**